tensorflow_datasets/text/wikipedia_dump.py

```diff
@@ -0,0 +1,92 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Wikipedia Dump Dataset'''
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """\
+@misc{
+  author = {Ortman, Mike},
+  title = {wikipedia-dump},
+  year = {2018},
+  howpublished = {\\url{https://www.kaggle.com/mikeortman/wikipedia-sentences}}
+}\
+"""
+
+_DESCRIPTION = '''\
+Collection of 7.8 million sentences (one per line) from August 2018 English wikipedia dump. 
+'''
+
+class WikipediaDump(tfds.core.GeneratorBasedBuilder):
+  '''Wikipedia dump dataset builder'''
+
+  VERSION = tfds.core.Version('1.0.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder = self,
+        description = _DESCRIPTION,
+        features = tfds.features.FeaturesDict({'text': tfds.features.Text()}),
+        supervised_keys = None,
+        homepage = 'https://www.kaggle.com/mikeortman/wikipedia-sentences',
+        citation = _CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    '''RETURNS SplitGenerators'''
+    download_path = dl_manager.download_and_extract('https://www.kaggle.com/mikeortman/wikipedia-sentences/download/PgEEhE0Qepk09CFW4P6a%2Fversions%2FkOLDVvHsx79ZASdBRIKR%2Ffiles%2Fwikisent2.txt?datasetVersionNumber=3')
+    if tf.io.gfile.isdir(download_path):
+      txt_path = os.path.join(download_path, 'input.txt')
+    else:
+      txt_path = download_path
+    with tf.io.gfile.GFile(txt_path, 'r') as f:
+      text = f.read()
+
+    # 80/10/10 split
+    i = int(len(text) * 0.8)
+    train_text = text[:i], text[i:]
+    i = int(len(text) * 0.5)
+    validation_text, text = text[:i], text[i:]
+    test_text = text
+
+    return [
+            tfds.core.SplitGenerator(
+                name = tfds.Split.TRAIN,
+                gen_kwargs = {
+                    'split_key' : 'train',
+                    'split_text' : train_text
+                },
+            ),
+            tfds.core.SplitGenerator(
+                name = tfds.Split.VALIDATION,
+                gen_kwargs = {
+                    'split_key' : 'validation',
+                    'split_text' : validation_text,
+                },
+            ),
+    ]
+
+    def _generate_examples(self, split_key, split_text):
+      data_key = split_key 
+      feature_dict = {'text' : split_text}
+      yield data_key, feature_dict
```

Comment:
This doesn't look like a good way to deal with stuffs.
Here are the steps which you should follow.
1. Read the data iteratively (Since allocating everything in the memory is not always a good idea)
 2. Yield each sentence instead of yielding everything. (Afterall it's wikipedia "sentences" dataset)

Reply:
Yielding needs an unique key.
so one way of dealing with things is to get an index using `enumerate()` and yield the index along with the line.
```python3
for index, text in enumerate(<iterator that returns one sentence at a time>):
  yield index, {"text": text}
```